algoliasearch-core/src/main/java/com/algolia/search/SearchClientBase.java

```diff
@@ -12,4 +12,8 @@ void waitTask(
       @Nonnull String indexName, long taskId, long timeToWait, RequestOptions requestOptions);
 
   void waitTask(@Nonnull String indexName, long taskId);
+
+  void waitAppTask(long taskId, long timeToWait, RequestOptions requestOptions);
```

Comment:
Introducing `waitAppTask` to mark the difference between waiting for an indexing task vs application task.
The objective is to avoid misuse of this method for index task.